ehttp/src/types.rs

```diff
@@ -28,6 +26,20 @@ impl Request {
         }
     }
 
+    /// Get `Header` list from request with the given key.
+    pub fn get_header(&self, key: String) -> Vec<String> {
+        self.headers
+            .iter()
+            .filter(|h| h.0 == key)
+            .map(|h| h.1.clone())
+            .collect()
+    }
```

Comment:
This could return `Vec<&str>` or `impl Iterator<Item = &str>`.

The `key` should be `&str`

Also `header_values` is a better name (rust API:s don't use the `get_` prefix for getters).


```suggestion
    /// Get `Header` list from request with the given key.
    pub fn header_values(&self, key: &str) -> impl Iterator<Item = &str> {
        self.headers
            .iter()
            .filter(|(name, _)| name == key)
            .map(|(_, value)| value.as_str())
    }
    
    /// Get the first matching header value for the given key.
    pub fn header_value(&self, key: &str) -> Option<&str> {
        self.header_values(key).next()
    }
```